analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ProgramState.cs

```diff
@@ -210,5 +211,22 @@ private ProgramState ResetFieldConstraints(Func<IFieldSymbol, bool> predicate)
             }
             return state;
         }
+
+        [Conditional("DEBUG")]
+        public void CheckStateConsistency()
+        {
```

Comment:
I used this helper to find any inconsistencies in the states. It is only executed locally in DEBUG builds because in CI we use the release builds. I'd like to keep it and ideally we should also run these checks in CI by producing also an DEBUG build. I have created a [Trello card](https://trello.com/c/Dqj90LlS) to start a discussion if this is valuable to have.